RecoMET/METAlgorithms/src/METSignificance.cc

```diff
@@ -93,8 +94,7 @@ metsig::METSignificance::getCovariance(const edm::View<reco::Jet>& jets,
 
    }
 
-   // calculate sumPt
-   double sumPt = 0;
+   // calculate sumPtUnclustered
```

Comment:
I think you should keep `sumPtUnclustered = 0;` since you are supposing `sumPtUnclustered` is equal to zero.
Otherwise, calling N times `getMETCovMatrix(iEvent, iSetup, sumPtUnclustered);` would give you `sumPtUnclustered` x N

Reply:
In the current scheme there is no possibility to call it multiple times:
```C++
    //add the MET significance
    if(calculateMETSignificance_) {
      double sumPtUnclustered=0;
      const reco::METCovMatrix& sigcov = getMETCovMatrix(iEvent, iSetup, sumPtUnclustered);
      amet.setSignificanceMatrix(sigcov);
      double metSig=metSigAlgo_->getSignificance(sigcov, amet);
      amet.setMETSignificance(metSig);
      amet.setMETSumPtUnclustered(sumPtUnclustered);
    }
```

I agree that for possible future usages different from what was designed here having such an initialization at 0 of ```sumPtUnclustered``` inside ```metsig::METSignificance::getCovariance()``` may represent an extra safety belt.

@silviodonato should we restrict that extra safety fix to the master version of this code, in view of possible future developments, or do you want it backported also to 10_6 (and 10_2 as well, where this code was already merged by you in the present form with #29276)?

Reply:
We can merge this PR as it is, then please make a fix for master, 10_6 and 10_2 (clearly it is not urgent). Having the fix only in master might be dangerous in case of backports